packages/block-editor/src/components/link-control/index.js

```diff
@@ -249,16 +314,26 @@ function LinkControl( {
 			}
 		);
 
-		const manualLinkEntryTypes = [ 'url', 'mailto', 'tel', 'internal' ];
+		const directLinkEntryTypes = [ 'url', 'mailto', 'tel', 'internal' ];
+		const isSingleDirectEntryResult =
+			suggestions.length === 1 &&
+			directLinkEntryTypes.includes(
+				suggestions[ 0 ].type.toLowerCase()
+			);
+		const shouldShowCreateEntity =
+			createEntity &&
+			! isSingleDirectEntryResult &&
+			! isInitialSuggestions;
+
+		// According to guidelines aria-label should be added if the label
+		// itself is not visible.
+		// See: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role
 		const searchResultsLabelId = isInitialSuggestions
 			? `block-editor-link-control-search-results-label-${ instanceId }`
 			: undefined;
 		const labelText = isInitialSuggestions
 			? __( 'Recently updated' )
 			: sprintf( __( 'Search results for %s' ), inputValue );
-		// According to guidelines aria-label should be added if the label
-		// itself is not visible.
-		// See: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role
```

Comment:
It seemed the comment was more accurately placed prior to this change.

Separately: I noticed yesterday that we use `aria-labelledby` with a corresponding paragraph element, instead of `aria-label`. I'm not sure why we don't just use `aria-label`, when the paragraph is visually hidden anyways? My understanding of `aria-labelledby` is that it's appropriate when there is another _visible_ element which can serve as the equivalent purpose of `aria-label`.

Reply:
@jeryj Any chance that given your a11y experience you could resolve this?

Reply:
I'll do my best 👍 This one always gets me mixed up. Maybe this time around it'll stick in my mind 😂

Reply:
For my own notes and reference, spec says: 

> If the interface is such that it is not possible to have a visible label on the screen, authors SHOULD use aria-label and SHOULD NOT use aria-labelledby.
https://www.w3.org/TR/wai-aria/#aria-label 

No visible label: Use `aria-label="The Label Text"`
Visible label: Use `aria-labelledby="#id-of-label"`

Reply:
I changed this implementation to reflect how I understand this usage. It seems that `aria-label` and `aria-labelledby` go on the same element. If there is no visible label, the `aria-label` gets placed directly on the search results box, not on the hidden label.

In this instance, I removed the hidden label entirely and placed `aria-label` on the search results box.